server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/DomainMappingsRoutes.java

```diff
@@ -114,28 +126,56 @@ public HaltException removeDomainMapping(Request request, Response response) {
             @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
                     message = "Internal server error - Something went bad on the server side.")
     })
-    public Map<String, List<String>> get(Request request, Response response) throws RecipientRewriteTableException {
+    public Map<String, List<String>> getAllMappings(Request request, Response response) throws RecipientRewriteTableException {
         return recipientRewriteTable.getAllMappings()
                 .entrySet()
                 .stream()
-                .collect(toImmutableMap(e -> e.getKey().getFixedDomain(),
-                        e -> e.getValue()
-                                .select(Mapping.Type.Domain)
-                                .asStream()
-                                .map(Mapping::asString)
-                                .map(Mapping.Type.Domain::withoutPrefix)
-                                .collect(toImmutableList())
+                .filter(mappingsEntry -> !mappingsEntry.getValue().isEmpty())
+                .filter(mappingsEntry -> mappingsEntry.getValue().contains(Mapping.Type.Domain))
+                .collect(Guavate.toImmutableMap(
```

Comment:
You can use Guavate:: toImmutableMultimap, make the return type of this method a multimap.

If you rebase on master, you will have Guava objects support (we added it very recently).

Reply:
The branch is on top of master and there is no method named `Guavate#toImmutableMultimap`.